### PR TITLE
feat(kit): `useNitro()` utility

### PR DESCRIPTION
**Pull request by pi0**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Taling with @danielroe, it is sometimes required to access nitro instance mainly to access resolved configuration. But it is tricky because nitro is intentionally initialized after all modules to reflect configuration changes by them.

This PR exposes `useNitro()` to be used within or after `ready` hook:

**Note:** Users should not **modify** the nitro configuration using this instance either. Because we have more than once instance (one is used for prerendering) and also nitro does not reflects configuration changes always after initializing. It is a read-only instance.

```ts
import { useNitro } from '@nuxt/kit'
nuxt.hook('ready', () => {
  console.log(useNitro())
})
```

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/kit/src/nitro.ts [hunk 1]
```diff
@@ -46,3 +46,26 @@ export function addServerHandler (handler: NitroEventHandler) {
...
+ * })
+ * ```
+ */
+export function useNitro (): Nitro {
+  const nuxt = useNuxt()
+  if (!nuxt._nitro) {
+    throw new Error('Nitro is not initialized yet. You can call `useNitro()` only after `ready` hook.')
+  }
+  return nuxt._nitro
+}
```
[the remark this answers is not in the document]
Comment (pi0) [reply packages/kit/src/nitro.ts:71]: I'm thinking to drop instance support for rest of composables too. We always have one (and only one) global nuxt instance.

[the remark this answers is not in the document]
Comment (danielroe) [reply packages/kit/src/nitro.ts:71]: Okay, fair point 👍

**Review by danielroe — APPROVED** [review]

File: packages/schema/src/types/nuxt.ts [hunk 2]
```diff
@@ -1,12 +1,14 @@
 import type { Hookable } from 'hookable'
 import type { Ignore } from 'ignore'
+import type { Nitro } from 'nitropack'
```
Comment (danielroe) [inline packages/schema/src/types/nuxt.ts:3]: Might need to add devDependency to schema.

Comment (pi0) [reply packages/schema/src/types/nuxt.ts]: https://github.com/nuxt/framework/pull/7557/commits/b95a8f0694a538b71e9a617f65edc156074e7a8e

**Pull request merged**